interfaces/matlab_experimental/Thermo/ThermoPhase.m

```diff
@@ -0,0 +1,1460 @@
+        %
+        % v = tp.minTemp
+        %
+        % The parameterizations used to represent the temperature-dependent
+        % species thermodynamic properties are generally only valid in some
+        % finite temperature range, which may be different for each species
+        % in the phase.
+        %
+        % See also: :mat:class:`maxTemp`
+        %
+        % :param tp:
+        %     Instance of class :mat:class:`ThermoPhase` (or another
+        %     object that derives from ThermoPhase)
+        % :return:
+        %     Vector of minimum temperatures of all species
+        minTemp
+
+        refPressure % Reference pressure for standard-state. Units: Pa.
+
+        % Speed of sound.
+        %
+        % c = tp.soundspeed
+        %
+        % If the phase is an ideal gas, the speed of sound is calculated by:
+        %
+        % .. math:: c = \sqrt{\gamma * R * T}
+        %
+        % where :math:`\gamma` is the ratio of specific heats, :math:`R` is
+        % the specific gas constant, and :math:`T` is the temperature. If the
+        % phase is not an ideal gas, the speed of sound is calculated by
+        %
+        % .. math:: c = \sqrt{\left(\frac{\partial p}{\partial \rho}\right)_s}
+        %
+        % where :math:`p` is the pressure and :math:`\rho` is the density,
+        % and the subscript :math:`s` indicates constant entropy. This is
+        % approximated by slightly increasing the density at constant entropy
+        % and computing the change in pressure.
+        %
+        % .. math:: c = \sqrt{\frac{p_1 - p_0}{\rho_1-\rho_0}}
+        %
+        % :param tp:
+        %     Instance of class :mat:class:`ThermoPhase` (or another
+        %     class derived from ThermoPhase)
+        % :return:
+        %     The speed of sound. Units: m/s
+        soundSpeed
+
+        thermalExpansionCoeff % Thermal expansion coefficient. Units: 1/K.
+
+        vaporFraction % Vapor fraction of the phase.
+
+    end
+
+    properties (Dependent = true)
+        V % Specific volume. Units: m^3/kg.
+        DP
+        DPX
+        DPY
+        HP
+        HPX
+        HPY
+        PV
+        PVX
+        PVY
+        SH
+        SHX
+        SHY
+        SP
+        SPX
+        SPY
+        ST
+        STX
+        STY
+        SV
+        SVX
+        SVY
+        TD
+        TDX
+        TDY
+        TH
+        THX
+        THY
+        TP
+        TPX
+        TPY
+        TV
+        TVX
+        TVY
+        UV
+        UVX
+        UVY
+        UP
+        UPX
+        UPY
+        VH
+        VHX
+        VHY
+    end
+
+    methods
+        %% ThermoPhase Class Constructor
+
+        function tp = ThermoPhase(src, id)
+            % Create a :mat:class:`ThermoPhase` object.
+            ctIsLoaded;
+
+            if nargin ~= 2
+                error('ThermoPhase expects 2 input arguments.');
+            end
+
+            tp.tpID = ctFunc('thermo_newFromFile', src, id);
```

Comment:
I set `tpID` to be `SetAccess = protected` because right now the `Solution` class constructor calls the `ThermoPhase` class constructor twice. One is to inherit all `ThermoPhase` properties and methods into the `Solution` class. The other is to create the property `Solution.tp` which is used later as an input argument when the `Kinetics` and `Transport` class constructors are called. This is because of the following restrictions in Matlab:
```
The subclass constructor must make all calls to superclass constructors before any other references to the object (obj). This restriction includes assigning property values or calling ordinary class methods. Also, a subclass constructor can call a superclass constructor only once.
```
Because of this duplication, I have to set `Solution.tpID` to be the same as `Solution.tp.tpID` at the end of the `Solution` class constructor. The write access is still limited to methods within `ThermoPhase` and `Solution` classes only and will not be modified by other code.

Reply:
Is the unused `ThermoPhase` in the C++ core deleted? 

Hopefully this can be replaced before long once the C++ `Solution` becomes available in the `clib` interface (see Cantera/enhancements#158).

Reply:
Yes, `Solution` and `Interface` class constructors now call the destructor for `ThermoPhase` at the end so it gets deleted.